src/JuMP.jl

```diff
@@ -943,15 +953,63 @@ function Base.getindex(m::JuMP.AbstractModel, name::Symbol)
 end
 
 """
-    Base.setindex!(m::JuMP.Model, value, name::Symbol)
+    Base.setindex!(model::JuMP.AbstractModel, value, name::Symbol)
+
+stores the object `value` in the model `model` using so that it can be accessed
+via `getindex`.  Can be called with `[]` syntax.
+"""
+function Base.setindex!(model::AbstractModel, value, name::Symbol)
+    object_dictionary(model)[name] = value
+end
+
+"""
+    delete(model::JuMP.AbstractModel, name::Symbol)
```

Comment:
I'm a bit worried about confusion from the overloaded meaning of `delete`. There's a big difference between `delete(m, x)` and `delete(m, :x)` that's understated by the 1-character difference. I'd tend to prefer a different name like `unregister` for simply deleting the key from the object dictionary (but could be convinced otherwise).

Reply:
To be sure we talk about the same thing, you mean `unregister` so that the user needs to do:
```julia
delete(m, x)
unregister(m, :x)
```

Reply:
Yes, I'm proposing `delete` and then an explicit `unregister`.

Reply:
@odow has a counter argument against `unregister` in https://github.com/jump-dev/JuMP.jl/pull/2232#issuecomment-622011209